share/doc/wake/basic_wake_tutorial.md

```diff
@@ -0,0 +1,628 @@
+```
+def aveM = \x\y (x+y) % 2
+```
+
+* Function duplicate concantenates or copies a given string twice
+```
+def duplicate s  ="{s}{s}"
+```
+
+* integerToUnicode function takes the ASCII input and provides its equivalent character
+```
+wake -x 'integerToUnicode 99'
+```
+
+* unicodeToInteger function gives the ASCII value of the given character
+```
+wake -x 'unicodeToInteger "a"'
+```
+
+* strlen function returns the length of the given string
+```
+wake -x 'strlen "abcde"'
+```
+
+* Below function takes a string and returns the middle character of the string
+```
+def middle fullstring =
+  def len = if (strlen fullstring % 2 == 0) then (strlen fullstring / 2) else  ((strlen fullstring / 2) + 1)
+  def outputstr = explode fullstring | take len | reverse | head 
+  outputstr
+```
+
+* Below function takes a string and returns the string removing the first and last character
+```
+def dtrunc fullstring =
+  def len = strlen fullstring
+  def outputstr = explode fullstring | take (len-1) | tail | cat
+  outputstr
+```
+
+* Function incfirst takes a string and a character string and replaces the first character of the string with the given character string
+```
+def incfirst fullstring chrstr = explode fullstring | tail | prepend chrstr | cat
+```
+
+* Function reorder takes a string , split the string and then reverses the splitted string and then concantenates both (Ex: hangover -> overhang)
+```
+def reorder fullstring = 
+  def Pair firststr laststr = 
+    explode fullstring
+    | splitAt (strlen fullstring / 2) 
+  cat (laststr ++ firststr)                         
+```
+
+* Function dubmid takes a string, duplicates the middle character and gives out the string (Ex: hapy -> happy))
+```
+def dubmid fullstring =
+  def Pair firststr laststr = 
+    explode fullstring
+    | splitAt (strlen fullstring / 2)
+  def dupli = laststr | take 1
+  cat (firststr ++ dupli ++ laststr)
+```
+
+* Takes a string and then removes the last character in a string
+```
+wake -x '"been"| explode | take 3| cat'
+```
+
+* Replaces the first charcter with the character "c"
+```
+wake -x '"bad"| explode | drop 1| prepend "c" | cat' 
+```
+### Tuple
+A tuple is collection of items of different types. `get`,`set` and `edit` are the three inbuilt methods that can be used with the tuples.
+* Below example shows the declaration of a tuple. For more information about the tuple look into wake database
+```
+def abcd = 1;"xx";1e5;Unit
+```
+### List
+A List is a sequence of items of same type. Empty list returns `Nil`. `head` returns the first element in the list. `tail` returns the rest of the list without the `head`. 
+* Example shows how to prepend the list of integers with the given value
+```
+def d = 1,2,3,Nil
+def listprepend data1 = prepend data1 d
+```
+
+* Example creates a list of given elements
+```
+def fone xx = xx,xx,xx,Nil
+```
+* Try out these examples
+```
+wake -x '"south" | explode | take 1| cat'
```

Comment:
you have not yet explained the `|` operator which is a key wake concept

Reply:
I will add this in the next commit